libraries/abstractions/transport/utest/transport_secure_sockets_utest.c

```diff
@@ -0,0 +1,740 @@
+    invalidNetworkContext.tcpSocket = NULL;
+
+    bytesReceived = SecureSocketsTransport_Recv( NULL, networkBuffer, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesReceived );
+
+    bytesReceived = SecureSocketsTransport_Recv( &invalidNetworkContext, NULL, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesReceived );
+
+    bytesReceived = SecureSocketsTransport_Recv( &invalidNetworkContext, networkBuffer, 0 );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesReceived );
+
+    bytesReceived = SecureSocketsTransport_Recv( &invalidNetworkContext, networkBuffer, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesReceived );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test the happy path case when #SecureSocketsTransport_Recv is able to receive all
+ * expected bytes over the network stack successfully.
+ */
+void test_SecureSocketsTransport_Recv_All_Bytes_Received_Successfully( void )
+{
+    int32_t bytesReceived;
+
+    SOCKETS_Recv_ExpectAnyArgsAndReturn( BYTES_TO_RECV );
+    bytesReceived = SecureSocketsTransport_Recv( &networkContext, networkBuffer, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( BYTES_TO_RECV, bytesReceived );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Recv returns an error when #SOCKETS_Recv fails to
+ * receive data over the network stack.
+ */
+void test_SecureSocketsTransport_Recv_Network_Error( void )
+{
+    int32_t bytesReceived;
+
+    SOCKETS_Recv_ExpectAnyArgsAndReturn( SOCKETS_EWOULDBLOCK );
+
+    bytesReceived = SecureSocketsTransport_Recv( &networkContext, networkBuffer, BYTES_TO_RECV );
+
+    /* SSL_ERROR_WANT_READ implies there is no data to receive, so we expect
+     * that no bytes have been received. */
+    TEST_ASSERT_EQUAL( 0, bytesReceived );
+
+    SOCKETS_Recv_ExpectAnyArgsAndReturn( SECURE_SOCKETS_READ_WRITE_ERROR );
+    bytesReceived = SecureSocketsTransport_Recv( &networkContext, networkBuffer, BYTES_TO_RECV );
+
+    /* SSL_ERROR_WANT_READ implies there is no data to receive, so we expect
+     * that no bytes have been received. */
+    TEST_ASSERT_EQUAL( SECURE_SOCKETS_READ_WRITE_ERROR, bytesReceived );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test the happy path case when #SecureSocketsTransport_Recv but received less than BYTES_TO_RECV.
+ */
+void test_SecureSocketsTransport_Recv_Bytes_Received_Partially( void )
+{
+    int32_t bytesReceived;
+
+    SOCKETS_Recv_ExpectAnyArgsAndReturn( BYTES_TO_RECV - 1 );
+    bytesReceived = SecureSocketsTransport_Recv( &networkContext, networkBuffer, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( BYTES_TO_RECV - 1, bytesReceived );
+}
+
+
+/*-------------------------------------------------------------------*/
+/*-----------------------End Tests-----------------------------------*/
+/*-------------------------------------------------------------------*/
```

Comment:
Please add new lines at the end of source code files.

Reply:
fixed in 5c6afe0c590c76d8c90a6972c9798058702963f1

resolving this comment, feel free to let me know if need more work.